staging/src/k8s.io/apimachinery/pkg/conversion/converter.go

```diff
@@ -442,8 +442,27 @@ func (c *Converter) doConversion(src, dest interface{}, flags FieldMatchingFlags
 	if fn, ok := c.generatedConversionFuncs.untyped[pair]; ok {
 		return fn(src, dest, scope)
 	}
-	// TODO: consider everything past this point deprecated - we want to support only point to point top level
-	// conversions
+
+	if pair.source == pair.dest {
+		// TODO: Ideally, to avoid this hack, we should auto-generate self-conversions.
```

Comment:
include the EnforcePtr(src) and EnforcePtr(dest) checks here?

Reply:
done